### DISCUSSION TITLE
Styling via the UI

### DISCUSSION THREAD

**Discussion by lstillwe** [TerriaJS/terriajs#6659] [Q&A] [answered]
I have seen some discussions in issues and pull requests for the capability to style layers dynamically via the UI.
Specifically I am looking to change colormaps for raster type, WMS layers.
Is this a feature that is already available, or planned in the future? If not, is this easily implemented in custom code?

Thanks very much,
Lisa Stillwell.

**Comment by hardreddata** [comment, marked as answer]
Hi Lisa,

My understanding here is that the raster style would be baked into the WMS and cannot be changed at the Terria level.

You need to address this at the source of the WMS (eg: GeoServer). Certainly if the WMS publishes layers with a choice of styles you can access them from Terria.

Cheers.

**Reply by lstillwe** [reply]
Thank you!
